sql/analyzer/resolve_subqueries.go

```diff
@@ -57,3 +58,33 @@ func resolveSubqueryExpressions(ctx *sql.Context, a *Analyzer, n sql.Node, scope
 		return s.WithQuery(analyzed), nil
 	})
 }
+
+// cacheSubqueryResults determines whether it's safe to cache the results for any subquery expressions. Caching
+// subquery results is safe in the case that no outer scope columns are referenced.
+func cacheSubqueryResults(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope) (sql.Node, error) {
```

Comment:
Perhaps this isn't an issue and I'm completely off base, but wouldn't this cache a nondeterministic subquery such as `(SELECT FLOOR(RAND()*5) IN (1, 2, 3)`?

Reply:
No, you are totally correct. I'll add a TODO for this, nice catch.

Reply:
I went to fix this and add tests, and discovered that result caching was totally broken, so thanks for this!